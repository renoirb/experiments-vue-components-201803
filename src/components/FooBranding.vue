<template>
  <div :class="[classNames]">
    <a
      :href="productUrl"
    >
      {{ productName }}
    </a>
    &nbsp;
    &copy;
    &nbsp;
    {{ firstYear }}-{{ year }}
    &nbsp;
    <a
      :href="publisherUrl"
    >
      {{ publisherName }}
    </a>
  </div>
</template>

<script>
const classNames = ['hotpink--text', 'text-xs-center', 'pa-2'];

// @vue/component
export default {
  props: {
    productName: {
      type: String,
      default: 'CatFacts',
    },
    productUrl: {
      type: String,
      default: 'http://www.catfacts.co/',
    },
    publisherName: {
      type: String,
      default: 'PiedPiper',
    },
    publisherUrl: {
      type: String,
      default: 'http://www.piedpiper.com/',
    },
    firstYear: {
      type: Number,
      default: 1998,
    },
  },
  data: () => ({
    classNames,
  }),
  computed: {
    year: () => new Date().getFullYear(),
  },
};
</script>