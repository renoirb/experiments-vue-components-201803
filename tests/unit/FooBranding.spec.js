import { shallow } from '@vue/test-utils'
import Vue from 'vue'
import FooBranding from '@/components/FooBranding.vue'

describe('FooBranding.vue', () => {
  it('renders props.productName when passed', () => {
    const productName = 'sg59PgYRUqI79HW'
    const wrapper = shallow(FooBranding, {
      propsData: { productName }
    })
    expect(wrapper.text()).toMatch(productName)
    expect(wrapper.element).toMatchSnapshot()
  })
  it('Should have the following default markup', () => {
    // Thanks https://alexjoverm.github.io/2017/08/21/Write-the-first-Vue-js-Component-Unit-Test-in-Jest/
    let Component = Vue.extend(FooBranding) // Create a copy of the original component
    const vm = new Component({
      propsData: { // Replace data value with this fake data
        productName: 'SomethingElse'
      }
    }).$mount() // Instances and mounts the component
    expect(vm.$el).toMatchSnapshot()
  })
})
