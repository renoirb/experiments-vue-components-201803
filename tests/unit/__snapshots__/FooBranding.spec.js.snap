// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooBranding.vue Should have the following default markup 1`] = `
<div
  class="grey lighten-4 elevation-4"
>
  <a
    href="http://www.catfacts.co/"
  >
    
    SomethingElse
  
  </a>
  
   
  ©
   
  1998-2018
   
  
  <a
    href="http://www.piedpiper.com/"
  >
    
    PiedPiper
  
  </a>
</div>
`;

exports[`FooBranding.vue renders props.productName when passed 1`] = `
<div
  class="grey lighten-4 elevation-4"
>
  <a
    href="http://www.catfacts.co/"
  >
    
    sg59PgYRUqI79HW
  
  </a>
  
   
  ©
   
  1998-2018
   
  
  <a
    href="http://www.piedpiper.com/"
  >
    
    PiedPiper
  
  </a>
</div>
`;
